src/components/productGallery/ProductGallery.jsx

```diff
@@ -0,0 +1,74 @@
+import React, { useState } from "react";
+import "./ProductGallery.css";
+
+const ProductGallery = ({
+  bigImage,
+  smallTopLeftImage,
+  smallTopRightImage,
+  smallBottomLeftImage,
+  smallBottomRightImage,
+}) => {
+  const [showPopupProductGallery, setShowPopupProductGallery] = useState(true);
+  const showPopupHandler = () => {
+    setShowPopupProductGallery(!showPopupProductGallery);
+  };
```

Comment:
Using 
setShowPopupProductGallery((prevState) => !prevState); 
instead of 
setShowPopupProductGallery(!showPopupProductGallery); 
ensures we always toggle based on the latest state, which prevents issues if the button is clicked multiple times quickly.

Reply:
Great point. I will change to ((prevState) => !prevState)